lib/repo-type-mappings/yocto-based-OS-image/versionist.conf.js

```diff
@@ -49,11 +55,33 @@ module.exports = {
     return 'patch'
   },
   incrementVersion: (currentVersion, incrementLevel) => {
-    const revision = Number(currentVersion[currentVersion.length - 1])
-    if (!_.isFinite(revision)) {
-      throw new Error(`Could not extract revision number from ${currentVersion}`)
+    if (isESR(currentVersion)) {
+      return semver.inc(currentVersion, 'patch')
     }
-    return currentVersion.slice(0, currentVersion.length - 1) + (revision + 1)
+    const parsedCurrentVersion = semver.parse(currentVersion)
+    if ( parsedCurrentVersion.build && parsedCurrentVersion.build != '' ) {
```

Comment:
```suggestion
    if ( parsedCurrentVersion.build) {
```
nit, truthy will treat empty strings as false

Reply:
Addressed.

Reply:
Actually @klutchell what I didn't realize is that `parsedCurrentVersion.build` is an object type so it was being detected as empty with:
```
if ( parsedCurrentVersion.build && parsedCurrentVersion.build != '' )
```
But not with:
```
 if ( parsedCurrentVersion.build ) 
```

Reply:
See https://github.com/product-os/balena-versionist/pull/69